sdk/messaging/azservicebus/example_processor_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azservicebus_test
+
+import (
+	"context"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus"
+)
+
+var processor *azservicebus.Processor
+
+func ExampleClient_NewProcessorForQueue() {
+	processor, err = client.NewProcessorForQueue(
```

Comment:
This is purposeful (although I agree with you).

If you look at all the individual tests the reason they compile is because they're using global definitions of these variables that are outside of their tests (otherwise, each test would require boilerplate declartions, making them less useful).

My hope (which I never got around to testing) is that I can actually run these tests, with each test more or less just filling in the global variables they need (making so many assumptions).

Do you just include all the client boilerplate in each function's example?

Reply:
I did include the full client boilerplate for both KV and Tables. It's ugly but it does give the end user a fully runnable sample that can be copy-pasted and run right away. I think this helps with approachability in the long term

Reply:
Let's leave this for now, it might be better to move the full examples into the struct level example instead.